### DISCUSSION TITLE
pkg-config does not use my $PKG_CONFIG_PATH

### DISCUSSION THREAD

**Discussion by aacirino** [Homebrew/discussions#2008] [Casks]
Hi, in macOS Catalina 10.15.7 with up-to-date `brew` and `pkg-config` I am not able to make `pkg-config` use my `$PKG_CONFIG_PATH` variable exported in my `./~zshrc` profile. Using `pkg-config` to inspect what it is using gives me a very wrong and short path:

```
pkg-config --variable pc_path pkg-config
/usr/local/lib/pkgconfig:/usr/local/share/pkgconfig:/usr/lib/pkgconfig:/usr/local/Homebrew/Library/Homebrew/os/mac/pkgconfig/10.11
```
I can't find a way to make `pkg-config` use my path.

Any help is welcome,

Angelo

**Comment by gromgit** [comment]
> Using pkg-config to inspect what it is using gives me a very wrong and short path:

Nope, you're misreading the output. From the `pkg-config` man page:
> **ENVIRONMENT VARIABLES**
>        `PKG_CONFIG_PATH`
>               A colon-separated (on Windows, semicolon-separated) list of directories to search for .pc files.  ***The default directory will always be searched after searching the path***
> [...]
> **PKG-CONFIG DERIVED VARIABLES**
> [...]
>        `pc_path`
>               ***The default search path used by `pkg-config` when searching for .pc files.*** This can be used in a query for the `pkg-config` module itself itself:
>                 `$ pkg-config --variable pc_path pkg-config`

In other words, `pkg-config` will search the directories (if any) specified in `PKG_CONFIG_PATH`, *then* (if necessary) the directories listed in `pkg-config --variable pc_path pkg-config`. The latter is a fixed list that's configured at build time.

> I can't find a way to make `pkg-config` use my path.

Then you're setting it wrong. If `env | grep PKG_CONFIG_PATH` returns nothing...

**Comment by gromgit** [comment]
Proof that Homebrew `pkg-config` works as advertised on Catalina:
```
$ which pkg-config
/usr/local/bin/pkg-config

$ pkg-config --variable pc_path pkg-config
/usr/local/lib/pkgconfig:/usr/local/share/pkgconfig:/usr/lib/pkgconfig:/usr/local/Homebrew/Library/Homebrew/os/mac/pkgconfig/10.15

# Let's probe for system python
$ pkg-config --cflags python
Package python was not found in the pkg-config search path.
Perhaps you should add the directory containing `python.pc'
to the PKG_CONFIG_PATH environment variable
No package 'python' found

# Whoops, forgot PKG_CONFIG_PATH
$ export PKG_CONFIG_PATH=/System/Library/Frameworks/Python.framework/Versions/2.7/lib/pkgconfig

$ pkg-config --cflags python
-I/System/Library/Frameworks/Python.framework/Versions/2.7/include/python2.7
```

**Comment by aacirino** [comment]
This is my `PKG_CONFIG_PATH` as you suggested:
```~ env | grep PKG_CONFIG_PATH
PKG_CONFIG_PATH=/usr/local/lib/pkgconfig:/usr/local/opt/icu4c/lib/pkgconfig:/usr/local/opt/krb5/lib/pkgconfig:/usr/local/opt/libffi/lib/pkgconfig:/usr/local/opt/libressl/lib/pkgconfig:/usr/local/opt/curl/lib/pkgconfig:/usr/local/opt/qt/lib/pkgconfig:/usr/local/opt/tcl-tk/lib/pkgconfig:/usr/local/opt/openblas/lib/pkgconfig:/Users/aacirino/git/restclient-cpp:/usr/local/opt/zlib/lib/pkgconfig:/usr/local/opt/libxml2/lib/pkgconfig:/usr/local/hpx/lib/pkgconfig
```
But nevermind, after restarting my computer it is working. I don't understand what was wrong. Thanx and have a nice day.
